frame/sudo/src/mock.rs

```diff
@@ -0,0 +1,185 @@
+// Copyright 2020 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Test utilities
+
+use super::*;
+use frame_support::{
+	impl_outer_origin, impl_outer_dispatch, impl_outer_event, parameter_types,
+	weights::{Weight, DispatchClass}
+};
+use sp_core::H256;
+// The testing primitives are very useful for avoiding having to work with signatures
+// or public keys. 
+use sp_runtime::{Perbill, traits::{BlakeTwo256, IdentityLookup}, testing::Header};
+use sp_io;
+use crate as sudo;
+
+// Logger module to track execution.
+pub mod logger {
+	use super::*;
+	use frame_system::ensure_root;
+
+	pub trait Trait: system::Trait {
+		type Event: From<Event<Self>> + Into<<Self as system::Trait>::Event>;
+	}
+
+	decl_storage! {
+		trait Store for Module<T: Trait> as Logger {
+			AccountLog get(fn account_log): Vec<T::AccountId>;
+			I32Log get(fn i32_log): Vec<i32>;
+		}
+	}
+
+	decl_event! {
+		pub enum Event<T> where AccountId = <T as frame_system::Trait>::AccountId {
+			AppendAccount(AccountId, i32, Weight),
+			AppendI32(i32, Weight),
+		}
+	}
+
+	decl_module! {
+		pub struct Module<T: Trait> for enum Call where origin: <T as system::Trait>::Origin {
+			fn deposit_event() = default;
+
+			#[weight = FunctionOf(
+				|args: (&i32, &Weight)| *args.1,
+				DispatchClass::Normal,
+				Pays::Yes,
+			)]
+			fn privileged_i32_log(origin, i: i32, weight: Weight){
+				// Ensure that the `origin` is `Root`.	
+				ensure_root(origin)?;
+				<I32Log>::append(i);
+				Self::deposit_event(RawEvent::AppendI32(i, weight));
+			}
+
+			#[weight = FunctionOf(
+				|args: (&i32, &Weight)| *args.1,
+				DispatchClass::Normal,
+				Pays::Yes,
+			)]
+			fn non_privileged_i32_log(origin, i: i32, weight: Weight){
+				// Ensure that the `origin` is some signed account.		
+				ensure_signed(origin)?;
+				<I32Log>::append(i);
+				Self::deposit_event(RawEvent::AppendI32(i, weight));
+			}
+
+			#[weight = FunctionOf(
+				|args: (&i32, &Weight)| *args.1,
+				DispatchClass::Normal,
+				Pays::Yes,
+			)]
+			fn non_privileged_account_log(origin, i: i32, weight: Weight) {
+				// Ensure that the `origin` is some signed account.	
+				let sender = ensure_signed(origin)?;
+				<AccountLog<T>>::append(sender.clone());
+				Self::deposit_event(RawEvent::AppendAccount(sender, i, weight));
```

Comment:
This could just be a single function.

```
			fn non_privileged_i32_log(origin, i: i32, weight: Weight){
				// Ensure that the `origin` is some signed account.		
				ensure_signed(origin)?;
				<I32Log>::append(i);
				<AccountLog<T>>::append(sender.clone());
				Self::deposit_event(RawEvent::AppendI32(i, weight));
			}
```